src/Extensions/hcox_dustdead_mod.F

```diff
@@ -5638,74 +5638,149 @@ SUBROUTINE InstRemove ( Instance )
       ! Instance-specific deallocation
       IF ( ASSOCIATED(Inst) ) THEN
 
-      !-----------------------------------------------------------------
-      ! Nullify pointers
-      !-----------------------------------------------------------------
-      IF(ASSOCIATED( Inst%ERD_FCT_GEO  )) DEALLOCATE(Inst%ERD_FCT_GEO  )
-      IF(ASSOCIATED( Inst%SRCE_FUNC    )) DEALLOCATE(Inst%SRCE_FUNC    )
-      IF(ASSOCIATED( Inst%LND_FRC_DRY  )) DEALLOCATE(Inst%LND_FRC_DRY  )
-      IF(ASSOCIATED( Inst%MSS_FRC_CACO3)) DEALLOCATE(Inst%MSS_FRC_CACO3)
-      IF(ASSOCIATED( Inst%MSS_FRC_SND  )) DEALLOCATE(Inst%MSS_FRC_SND  )
-      IF(ASSOCIATED( Inst%SFC_TYP      )) DEALLOCATE(Inst%SFC_TYP      )
-      IF(ASSOCIATED( Inst%VAI_DST      )) DEALLOCATE(Inst%VAI_DST      )
+         !--------------------------------------------------------------
+         ! Deallocate fields of Inst before popping off from the list
+         ! in order to avoid memory leaks (Bob Yantosca (17 Aug 2022)
+         !--------------------------------------------------------------
+         IF ( ASSOCIATED( Inst%ERD_FCT_GEO ) ) THEN
+            DEALLOCATE(Inst%ERD_FCT_GEO  )
+         ENDIF
+         Inst%ERD_FCT_GEO => NULL()
 
-      !-----------------------------------------------------------------
-      ! Cleanup module arrays
-      !-----------------------------------------------------------------
+         IF ( ASSOCIATED( Inst%SRCE_FUNC ) ) THEN
+            DEALLOCATE(Inst%SRCE_FUNC )
+         ENDIF
+         Inst%SRCE_FUNC => NULL()
+
+         IF ( ASSOCIATED( Inst%LND_FRC_DRY ) ) THEN
+            DEALLOCATE(Inst%LND_FRC_DRY )
+         ENDIF
+         Inst%LND_FRC_DRY   => NULL()
+
+         IF ( ASSOCIATED( Inst%MSS_FRC_CACO3 ) ) THEN
+            DEALLOCATE(Inst%MSS_FRC_CACO3)
+         ENDIF
+         Inst%MSS_FRC_CACO3 => NULL()
+
+         IF ( ASSOCIATED( Inst%MSS_FRC_SND ) ) THEN
+            DEALLOCATE(Inst%MSS_FRC_SND )
+         ENDIF
+         Inst%MSS_FRC_SND => NULL()
+
+         IF ( ASSOCIATED( Inst%SFC_TYP ) ) THEN
+            DEALLOCATE(Inst%SFC_TYP )
+         ENDIF
+         Inst%SFC_TYP => NULL()
+
+         IF ( ASSOCIATED( Inst%VAI_DST ) ) THEN
+            DEALLOCATE(Inst%VAI_DST )
+         ENDIF
+         Inst%VAI_DST => NULL()
+
+! These are not used
```

Comment:
Could these be deleted if not used?

Reply:
Now removed in commit 758326e3d7886ff037502d871ac5c86b7ff6b1.  Changelog has been updated accordingly.